lib/facter/resolvers/selinux.rb

```diff
@@ -13,30 +13,28 @@ def post_resolve(fact_name, _options)
         end
 
         def retrieve_facts(fact_name)
-          mountpoint = read_mounts_file if @fact_list[:enabled].nil?
+          mountpoint = selinux_mountpoint
+
+          @fact_list[:enabled] = !mountpoint.empty? && read_selinux_config
```

Comment:
if `mountpoint.empty?` is true, it will never execute `read_selinux_config`, is this intended?

Reply:
Yes, `mountpoint.empty?` translates to selinuxfs not mounted, so there's no point to check for the config file (from what I understand). This is similar to the facter 3 implementation: https://github.com/puppetlabs/facter/commit/125a79e4da408bb4d4a86ebb7dd71c0ca27e288f#diff-c69186a8ef3e4d86f44def025a4e1bf8f132faa49d31b0e0b90adf55333fe325R70

Reply:
You are right. Filling @fact_list in various places is not easy reading for me.